examples/capture/main.rs

```diff
@@ -117,7 +117,7 @@ async fn run() {
     queue.submit(Some(command_buffer));
 
     // Note that we're not calling `.await` here.
-    let buffer_future = output_buffer.map_async(wgpu::MapMode::Read, 0, wgt::BufferSize::WHOLE);
+    let buffer_future = output_buffer.map_async(wgpu::MapMode::Read, ..);
```

Comment:
didn't you want to have it on the `BufferSlice`? what made you to reconsider?

Reply:
Oh, yeah, we can do that. Some of my previous attempts didn't work on `BufferSlice`, so I forgot to put it there this time. Should `get_mapped_range` and `get_mapped_range_mut` be on `BufferSlice` as well? Also, should `BufferSlice` have a `slice` method?

Reply:
That being said, I'm a little worried that moving the methods over to `BufferSlice` could obscure their usage a little.

Reply:
Since we can do this, and since the relative frame of reference for both `get_mapped_range*` and `map_async` is the same (beginning of the buffer), it makes sense to me that we'd have all of them on `BufferSlice`.

> Also, should BufferSlice have a slice method?

Hah, we could have that, yes, if you feel motivated to do this :)

Reply:
> That being said, I'm a little worried that moving the methods over to BufferSlice could obscure their usage a little.

I share the concern... and I think it's fine, given that buffer mapping should be considered an advanced feature of the API. Most users are ok without mapping at all.